cdk/setup-tool

```diff
@@ -0,0 +1,252 @@
+                ;;
+        esac
+    done
+    
+    [[ -n "$setup_name" ]] && say "Deploying setup: ${setup_name}"
+    
+    CLI_COMMAND="npx ts-node --prefer-ts-exts src/bin/cli.ts ${args}"
+    
+    cdk deploy --all --require-approval never --app \""${CLI_COMMAND}"\"
+    ok_or_die "Failed to deploy"
+
+    [[ -n "$setup_name" ]] && say "Successfully deployed stacks for setup: ${setup_name}"
+}
+
+# Destroy command implementation
+run_destroy_command() {
+    local setup_name=""
+    local destroy_all=false
+    local destroy_certificate=false
+    local destroy_role=false
+    local destroy_instance=false
+    local filtered_args=()
+
+    # First argument is filtered args array
+    if [[ $# -gt 4 ]]; then
+        filtered_args=("${@:1:$#-4}")
+    fi
+    
+    # Last 4 arguments are the flags
+    destroy_all="${@: -4:1}"
+    destroy_certificate="${@: -3:1}" 
+    destroy_role="${@: -2:1}"
+    destroy_instance="${@: -1:1}"
+
+    # Extract setup name from filtered args
+    local i=0
+    while [[ $i -lt ${#filtered_args[@]} ]]; do
+        case "${filtered_args[$i]}" in
+            --setup-name)
+                setup_name="${filtered_args[$((i+1))]}"
+                validate_setup_name "$setup_name"
+                break
+                ;;
+        esac
+        ((i++))
+    done
+
+    CLI_COMMAND=\""npx ts-node --prefer-ts-exts src/bin/cli.ts ${filtered_args[@]}"\"
+
+    if [[ "$destroy_all" == true ]]; then
+        say "Destroying all stacks for setup: ${setup_name}"
+        cdk destroy --all --app "${CLI_COMMAND}"
+    else
+        say "Destroying specific stacks for setup: ${setup_name}"
+        
+        local stacks_to_destroy=()
+        # Build list of stacks to destroy
+        if [[ "$destroy_certificate" == true ]]; then
+            stacks_to_destroy+=("${setup_name}-CertificateStack")
+        fi
+        if [[ "$destroy_role" == true ]]; then
+            stacks_to_destroy+=("${setup_name}-RoleStack")
+        fi
+        if [[ "$destroy_instance" == true ]]; then
+            stacks_to_destroy+=("${setup_name}-InstanceStack")
+        fi
+
+        if [[ ${#stacks_to_destroy[@]} -eq 0 ]]; then
+            die "No stacks specified for destruction"
+        fi
+
+        # Join stack names with spaces
+        local stack_list="${stacks_to_destroy[*]}"
+        say "Executing: cdk destroy --app ${CLI_COMMAND} ${stack_list}"
+        cdk destroy --app "${CLI_COMMAND}" "${stack_list}"
+    fi
+    
+    ok_or_die "Failed to destroy stack(s)"
+    say "Successfully destroyed stack(s)"
+}
+
+parse_deploy_args() {
+    if [ -z "$2" ] || [ "$2" == "-h" ] || [ "$2" == "--help" ]; then
+        npx ts-node --prefer-ts-exts src/bin/cli.ts -h
+        die -c 0
+    fi
+    run_deploy_command "$@"
+}
+
+parse_destroy_args() {
+    local filtered_args=()
+    local setup_name=""
+    local destroy_all=false
+    local destroy_certificate=false
+    local destroy_role=false
+    local destroy_instance=false
```

Comment:
nit: Is the `destroy_all` flag not redundant? We could just set the three specific flags and have less hassle to recover the flags in `run_destroy_command`. It should also make that function less complex having one flag less to handle.